worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -371,17 +371,34 @@ function isEmptyMap(mapValue) {
     return jsonString === '{}';
 }
 
+function searchFieldsForValue(fieldValues, value) {
+    for (var fieldValue of fieldValues) {
```

Comment:
We need to swap the loops.  The `for (var fieldValue of fieldValues) {` should be the outer loop as there will likely only be one iteration of it.

Reply:
[workflow-control_sample.txt](https://github.com/CAFDataProcessing/worker-workflow/files/9310203/workflow-control_sample.txt)

@dermot-hardy Something like this?

Removed the for loop from searchFieldsForValue and put it into each function separately - is an outer loop for fieldHasAnyStringValue function as it also has loop for multiple values

Reply:
Basically yes though that still needs some refinement.  The `.getValues()` will never return `null` so those checks can be removed.